src/Result.php

```diff
@@ -0,0 +1,183 @@
+<?php
+declare(strict_types=1);
+
+namespace FastRoute;
+
+use ArrayAccess;
+use RuntimeException;
+
+/**
+ * Result Object
+ *
+ * @implements ArrayAccess<int, mixed>
+ */
+class Result implements ArrayAccess
+{
+    public const NOT_FOUND = 0;
+    public const FOUND = 1;
+    public const METHOD_NOT_ALLOWED = 2;
+
+    /** @var bool */
+    protected $matched = false;
+
+    /** @var IRoute */
+    protected $route;
```

Comment:
This is a write-only property, why do we need it here?

Reply:
Because you can get the route object of the route that matched. I'm just waiting what happens to this PR to see if its worth to spend more of my time with this or not. If it's worth the time I'll continue and refactor the routes and make sure a route object is used everywhere.

Reply:
So let's break things down and provide a full implementation of something instead of half steps.

Reply:
The property already contains the route object, there is nothing to wait on.